spago-dev.nix

```diff
@@ -0,0 +1,78 @@
+{ system
+, lib
+, src
+, stdenv
+, slimlock
+, purix
+, purs
+, spago
+, nodejs
+, esbuild
+, python3
+, git
+, installShellFiles
+}:
+let
+  spagoBin = purix.buildSpagoPackages {
+    inherit src purs spago;
+    packages = [ "spago-bin" ];
+  };
+
+  npmDependencies = (slimlock.buildPackageLock { inherit src; }).overrideAttrs (x: {
+    nativeBuildInputs = (x.nativeBuildInputs or [ ]) ++ [ python3 ];
+  });
+in
+stdenv.mkDerivation {
+  name = "spago-dev";
+
+  version = "0.93.24";
+
+  meta = {
+    mainProgram = "spago";
+  };
+
+  nativeBuildInputs = [
+    purs
+    spago
+    python3
+    nodejs
+    git
```

Comment:
Yes. From the spago [README](https://github.com/purescript/spago#design-goals-and-reasons):

> Minimal dependencies: users s6hould not be expected to install a myriad of tools on their system to support various workflows. Spago only expects git and purs to be installed.

Reply:
It's needed because Spago uses git to interact with the registry repositories. If we only use spago in offline mode (ie. `spago build --offline`) then I believe the calls to fetch the registry will be skipped and we don't need `git`.